pykern/util.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+"""utils for pykern
+
+:copyright: Copyright (c) 2023 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+from pykern import pkconfig
+from pykern import pkinspect
+from pykern import pkio
+from pykern.pkdebug import pkdc, pkdexc, pkdlog, pkdp
+import os.path
+import sys
+
+
+_DEFAULT_ROOT = "run"
+
+
+def cfg_db_dir_parser(value):
+    """Config value or root package's parent or cwd with `_DEFAULT_ROOT`"""
+    if not os.path.isabs(value):
+        pkconfig.raise_error("must be absolute")
+    if not os.path.isdir(value):
+        pkconfig.raise_error("must be a directory and exist")
+    return pkio.py_path(value)
+
+
+def dev_run_dir(package_object):
+    """Initialize run dir
+
+    Args:
+        package_object (object): python object who's root_package will hold the run dir
+
+    Returns:
+        py.path.local: absolute path to run dir
+    """
+    assert pkconfig.in_dev_mode(), "run dir root must be configured except in dev"
+    r = (
+        pkio.py_path(
+            sys.modules[pkinspect.root_package(package_object)].__file__,
+        )
+        .dirpath()
+        .dirpath()
+    )
+    # Check to see if we are in our dev directory. This is a hack,
```

Comment:
This comment is not very good. I know I wrote it. Look for files at the root of a python project that are not installed when the package is installed. There are assumptions (above) about where this path is so we assume a particular file structure. This should be noted in the function's description.